corehq/apps/callcenter/sync_user_case.py

```diff
@@ -214,9 +219,31 @@ def _call_center_location_owner(user, ancestor_level):
 
 
 def sync_usercase(user):
+    helper = get_sync_usercase_helper()
+    if helper:
+        helper.commit()
+
+def get_sync_usercase_helper(user):
     if user.project.usercase_enabled:
-        _sync_user_case(
+        return _get_sync_user_case_helper(
             user,
             USERCASE_TYPE,
             user.get_id
         )
+
+def sync_user_cases(user):
+    """
+    Each time a CommCareUser is saved this method gets called and creates or updates
+    a case associated with the user with the user's details.
+
+    This is also called to create user cases when the usercase is used for the
+    first time.
+    """
+    with CriticalSection(get_sync_lock_key(user._id)):
+        helpers = [get_sync_usercase_helper(), get_call_center_case_helper()]
+        case_blocks = [h._case_block_to_submit for h in helpers if h]
+        case_blocks = [cb.as_text() for cb in case_blocks if cb]
+        submit_case_blocks(case_blocks, user.domain, device_id="sync_user_case")
+        for helper in helpers:
+            for task, task_args in helper._tasks_to_trigger:
+                task.delay(task_args))
```

Comment:
how about moving this to a function on the helper that can be called here and in `commit` that would keep it encapsulated in the class